pkg/koordlet/resmanager/blkio_reconcile.go

```diff
@@ -0,0 +1,476 @@
+			},
+		)
+		if err != nil {
+			klog.Errorf("%s: fail to update be class blkio config: %s", BlkIOReconcileName, err.Error())
+		} else {
+			klog.V(4).Infof("%s: reconcile be class blkio config finished", BlkIOReconcileName)
+		}
+	}
+	// root
+	if strategy.CgroupRoot != nil && strategy.CgroupRoot.BlkIOQOS != nil {
+		klog.V(4).Infof("%s: start to reconcile root class blkio config", BlkIOReconcileName)
+		blocks := []*slov1alpha1.BlockCfg{}
+		if *strategy.CgroupRoot.BlkIOQOS.Enable {
+			blocks = strategy.CgroupRoot.BlkIOQOS.Blocks
+		}
+		rootClassRelativePath := ""
+		rootClassPath := util.GetCgroupRootBlkIOAbsoluteDir()
+		err := b.updateBlkIOConfig(
+			blocks,
+			nil,
+			diskRecorder{
+				absolutePath:    rootClassPath,
+				getDiskRecorder: getDiskConfigRecorder,
+			},
+			blkioUpdater{
+				dynamicPath:    rootClassRelativePath,
+				getUpdaterFunc: getDiskConfigUpdaterFromBlockCfg,
+				getRemoverFunc: getDiskConfigRemoverFromDiskNumber,
+			},
+		)
+		if err != nil {
+			klog.Errorf("%s: fail to update root class blkio config: %s", BlkIOReconcileName, err.Error())
+		} else {
+			klog.V(4).Infof("%s: reconcile root class blkio config finished", BlkIOReconcileName)
+		}
+	}
+
+	// pods
+	podsMeta := b.resmanager.statesInformer.GetAllPods()
+	for _, podMeta := range podsMeta {
+		if extension.GetPodQoSClass(podMeta.Pod) == extension.QoSNone {
+			// ignore unknown qos pods
+			continue
+		}
+		if podMeta.Pod.Status.Phase != corev1.PodRunning {
+			klog.Warningf("%s: pod %s/%s is not in running status, ignoring...", BlkIOReconcileName, podMeta.Pod.Namespace, podMeta.Pod.Name)
+			continue
+		}
+		var err error
+		podBlkIOQoS := &slov1alpha1.BlkIOQOS{}
+		if podVolumeResult, exist := podMeta.Pod.Annotations[extension.AnnotationPodBlkioQoS]; exist {
+			if podVolumeResult != "" {
+				if podBlkIOQoS, err = parseBlkIOResult(podVolumeResult); err != nil {
+					klog.Errorf("%s: unmarshal pod annotation %v failed, error %v", BlkIOReconcileName, extension.AnnotationPodBlkioQoS, err)
+					continue
+				}
+			}
+		} else {
+			continue
+		}
+		klog.V(4).Infof("%s: start to reconcile pod %s/%s blkio config", BlkIOReconcileName, podMeta.Pod.Namespace, podMeta.Pod.Name)
+		podRelativePath := util.GetPodCgroupDirWithKube(podMeta.CgroupDir)
+		podPath := util.GetPodCgroupBlkIOAbsolutePath(podMeta.CgroupDir)
+		err = b.updateBlkIOConfig(
+			podBlkIOQoS.Blocks,
+			podMeta,
+			diskRecorder{
+				absolutePath:    podPath,
+				getDiskRecorder: getBlkIORecorder,
+			},
+			blkioUpdater{
+				dynamicPath:    podRelativePath,
+				getUpdaterFunc: getBlkIOUpdaterFromBlockCfg,
+				getRemoverFunc: getBlkIORemoverFromDiskNumber,
+			},
+		)
+		if err != nil {
+			klog.Errorf("%s: fail to update pod %s/%s blkio config: %s", BlkIOReconcileName, podMeta.Pod.Namespace, podMeta.Pod.Name, err.Error())
+		} else {
+			klog.V(4).Infof("%s: reconcile pod %s/%s blkio config finished", BlkIOReconcileName, podMeta.Pod.Namespace, podMeta.Pod.Name)
+		}
+	}
+}
+
```

Comment:
could diskRecorder and blkioUpdater be merged into one struct

Reply:
done